test/compatibility/test_wt10533.py

```diff
@@ -0,0 +1,153 @@
+        '''
+
+        import wiredtiger
+
+        conn = wiredtiger.wiredtiger_open('.', 'create,' + self.conn_config)
+        session = conn.open_session()
+        
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        # Create and populate a table.
+        session.create(self.uri, self.create_config)
+
+        c = session.open_cursor(self.uri)
+        for i in range(1, self.num_initial_rows):
+            c[i] = 'i' + str(i)
+        c.close()
+
+        # Create a lot of checkpoints to ensure that the checkpoint's "time" advances far.
+        # If there is already a checkpoint with a given timestamp, WiredTiger adds another second,
+        # and will keep adding more seconds, until it finds a unique timestamp that is larger than
+        # any existing timestamp.
+        for i in range(1, self.num_checkpoints):
+            c = session.open_cursor(self.uri)
+            c[self.num_initial_rows + i] = 'c' + str(i)
+            c.close()
+            session.checkpoint()
+
+        # Close
+        session.close()
+        conn.close()
+
+    def on_branch_test_checkpoint_downgrade(self):
+        '''
+        The second part of the test, which runs on a different branch.
+        '''
+        
+        import wiredtiger
+        
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        conn = wiredtiger.wiredtiger_open('.', self.conn_config)
+        session = conn.open_session()
+
+        # Start creating a backup.
+        backup_path = 'backup'
+        if os.path.exists(backup_path):
+            shutil.rmtree(backup_path)
+        os.mkdir(backup_path)
+        backup_cursor = session.open_cursor('backup:')
+
+        # Create a few checkpoints, each holding quite a bit of data.
+        for i in range(0, self.num_second_checkpoints):
+            c = session.open_cursor(self.uri)
+            for j in range(1, self.num_second_rows_per_checkpoint):
+                c[i * 1000000 + j] = 'x' + str(i) + '-' + str(j)
+            c.close()
+            session.checkpoint()
+
+        # Now actually create the backup.
+        while backup_cursor.next() == 0:
+            file_name = backup_cursor.get_key()
+            shutil.copy(file_name, os.path.join(backup_path, file_name))
+        backup_cursor.close()
+
+        session.close()
+        conn.close()
+
+        # And let's see if we can open the backup and verify that it has the right data.
+        conn = wiredtiger.wiredtiger_open(backup_path, self.conn_config)
+        session = conn.open_session()
+
+        c = session.open_cursor(self.uri)
+        for i in range(1, self.num_initial_rows):
+            v = 'i' + str(i)
+            if c[i] != v:
+                sys.stderr.write(f'Data mismatch on key {i}: expected {v}, got {c[i]}\n')
+                sys.exit(1)
+        c.close()
+
+        session.close()
+        conn.close()
+
+
+if __name__ == '__main__':
```

Comment:
We had a recent discussion about defining these lines: WT-11079. Are they necessary for this test?

Reply:
Thanks for the details @pmacko86. I have updated WT-11079 to make sure we don't keep this test intact!